test/lib/users/widgets/inst_root_first_test.rb

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env rspec
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "../../../test_helper"
+require "users/widgets/inst_root_first"
+require "cwm/rspec"
+
+describe Y2Users::Widgets::InstRootFirst do
+  subject(:widget) { described_class.new }
+
+  include_examples "CWM::CustomWidget"
+
+  before do
+    allow(Yast2::Popup).to receive(:show)
+  end
+
+  describe "#validate" do
+    let(:password?) { false }
+    let(:key?) { false }
+
+    let(:password_widget) do
+      instance_double(Users::PasswordWidget, empty?: !password?)
+    end
+
+    let(:public_key_selector) do
+      instance_double(Y2Users::Widgets::PublicKeySelector, empty?: !key?)
+    end
+
+    before do
+      allow(Users::PasswordWidget).to receive(:new).and_return(password_widget)
+      allow(Y2Users::Widgets::PublicKeySelector).to receive(:new).and_return(public_key_selector)
+    end
+
+    context "when neither a password nor public key was given" do
+      before do
+      end
+
+      it "returns false" do
+        expect(widget.validate).to eq(false)
+      end
+
+      it "displays an error" do
+        expect(Yast2::Popup).to receive(:show)
+          .with(/to provide at least a password/, headline: :error)
```

Comment:
We set the English locale during test, so OK. (https://github.com/yast/yast-users/blob/master/test/test_helper.rb#L25)